pkg/services/sqlstore/datasource.go

```diff
@@ -101,6 +103,14 @@ func AddDataSource(cmd *models.AddDataSourceCommand) error {
 			cmd.JsonData = simplejson.New()
 		}
 
+		if cmd.Uid == "" {
+			uid, err := generateNewDatasourceUid(sess, cmd.OrgId)
+			if err != nil {
+				return errutil.Wrapf(err, "Failed to generate Uid for Datasource %s", cmd.Name)
```

Comment:
I think it's more understandable if it's written UID (as the acronym for Unique ID), and I don't see any reason to capitalize datasource. It's also helpful I think to quote the name, since it might contain spaces.
```suggestion
				return errutil.Wrapf(err, "failed to generate UID for datasource %q", cmd.Name)
```